internal/recert/recert.go

```diff
@@ -96,3 +123,171 @@ func getIngressCNAndFile(certsFolder string) (string, string, error) {
+	// into a seed image.
+	log.Info("Run recert --force-expire tool")
+	recertConfigFile := path.Join(common.BackupDir, RecertConfigFile)
+	if err := createRecertConfigFileForSeedCreation(recertConfigFile); err != nil {
+		return fmt.Errorf("failed to create %s file", recertConfigFile)
+	}
+	if err := RunRecert(recertContainerImage, authFile, recertConfigFile, ops); err != nil {
+		return err
+	}
+
+	log.Info("Recert --force-expire tool ran and summary created successfully.")
+	return nil
+}
+
+func RestoreOriginalSeedCrypto(recertContainerImage, authFile string, ops ops.Ops) error {
+	log.Info("Running recert --extend-expiration tool to restore original seed crypto")
+
+	if err := RunUnauthenticatedEtcdServer(authFile, common.EtcdContainerName, ops); err != nil {
+		return err
+	}
+
+	defer func() {
+		log.Info("Killing the unauthenticated etcd server")
+		if _, err := ops.RunInHostNamespace("podman", "kill", "recert_etcd"); err != nil {
+			log.Errorf("Failed to kill recert_etcd container: %v", err)
+		}
+	}()
+
+	log.Info("Run recert --extend-expiration tool")
+	recertConfigFile := path.Join(common.BackupCertsDir, RecertConfigFile)
+	if err := createRecertConfigFileForSeedRestoration(recertConfigFile); err != nil {
+		return fmt.Errorf("failed to create %s file", recertConfigFile)
+	}
+	if err := RunRecert(recertContainerImage, authFile, recertConfigFile, ops); err != nil {
+		return err
+	}
+
+	log.Infof("Removing %s folder", common.BackupCertsDir)
+	if err := os.RemoveAll(common.BackupCertsDir); err != nil {
+		return fmt.Errorf("error removing %s: %w", common.BackupCertsDir, err)
+	}
+
+	log.Info("Certificates in seed SNO cluster restored successfully.")
+	return nil
+}
+
+func RunUnauthenticatedEtcdServer(authFile, name string, ops ops.Ops) error {
+	// Get etcdImage available for the current release, this is needed by recert to
+	// run an unauthenticated etcd server for running successfully.
+	etcdImage, err := getImageFromPodDefinition(common.EtcdStaticPodFile, common.EtcdStaticPodContainer)
+	if err != nil {
+		return err
+	}
+
+	log.Info("Run unauthenticated etcd server for recert tool")
+
+	command := "podman"
+	args := append(podmanRecertArgs,
+		"--authfile", authFile, "--detach",
+		"--name", name,
+		"--entrypoint", "etcd",
+		"-v", "/var/lib/etcd:/store",
+		etcdImage,
+		"--name", "editor", "--data-dir", "/store")
+
+	// Run the command and return an error if it occurs
+	if _, err := ops.RunInHostNamespace(command, args...); err != nil {
+		return err
+	}
+
+	log.Info("Waiting for unauthenticated etcd start serving for recert tool")
+	if err := waitForEtcd("http://" + common.EtcdDefaultEndpoint + "/health"); err != nil {
+		return fmt.Errorf("failed to wait for unauthenticated etcd server: %w", err)
+	}
+	log.Info("Unauthenticated etcd server for recert is up and running")
+
+	return nil
+}
+
+func getImageFromPodDefinition(podFile, containerName string) (string, error) {
```

Comment:
Maybe make it part of utils ? 
Another question is why we actually read it from file and not from pod itself? Though it is fine for me to leave it this way but maybe need another name?

Reply:
Ok, I prefer to keep it, will move it to utils as commented. 

Re naming, any preferred name, or suggestions?